### PR TITLE
Get container info from FactoryOS registry instead of cmdiag; Create service events for container status

### PR DESCRIPTION
**Pull request by spaceisfun**

### CODE REVIEW

**Comment by spaceisfun** [comment]
@microsoft/onecore any chance of a review by EOD here? Thanks!

File: src/Service/ServiceExe.cs [hunk 1]
```diff
@@ -1785,77 +1809,136 @@ private void RunTaskRunInContainer(ServerTaskRun hostRun)
...
-                var ipStrings = cmdiag.TaskOutput.Where(x => x.Contains("IPv4 address", StringComparison.InvariantCultureIgnoreCase));
-
-                foreach (var ipString in ipStrings)
-                {
-                    if (ipString != null)
-                    {
-                        ips.Add(IPAddress.Parse(ipString.Split(':', StringSplitOptions.RemoveEmptyEntries).Last().Trim()));
-                    }
+            {
+                // FactoryOS puts the container IP in SYSTEM\CurrentControlSet\Control\FactoryUserManager
```
Comment (hajya) [inline src/Service/ServiceExe.cs:1921]: hmm; i don't love this. I understand if its needed for now but we should file an item to track this (likely as a part of figuring out plans around container management)

**Review by hajya — CHANGES_REQUESTED** [review]
hmm. the logic looks fine, i'd love to avoid intertwining with internal or os src binaries/services where we can. Curious what you think.

File: build/internal/FactoryOrchestratorServiceTemplate.wm.xml [hunk 2]
```diff
@@ -13,6 +13,7 @@
       objectName="LocalSystem"
       start="auto"
       type="win32OwnProcess"
+      dependOnService="FactoryUserManager"
```
[the remark this answers is not in the document]
Comment (spaceisfun) [reply build/internal/FactoryOrchestratorServiceTemplate.wm.xml:16]: yeah youre right, this isnt needed

File: src/Service/ServiceExe.cs (hunk 1 above)
Comment (spaceisfun) [reply src/Service/ServiceExe.cs:1921]: The only options I know of currently are this or pinvoking a private API. I'm not sure if I can get this via HNS, if I can, that is JSON based and C# friendly, but I dont want to spend cycles on that now.

I'll make an issue

File: src/Service/ServiceExe.cs [hunk 3]
```diff
@@ -1494,14 +1510,18 @@ public FOService(ILogger<FOService> logger)
                 DisableUWPAppsPage = false;
                 DisableManageTasklistsPage = false;
                 DisableFileTransferPage = false;
-                DisableNetworkAccess = false;
+                DisableNetworkAccess = true;
 				EnableNetworkAccess = false;
```
[the remark this answers is not in the document]
Comment (spaceisfun) [reply src/Service/ServiceExe.cs:1514]: tabz

File: src/Service/ServiceExe.cs (hunk 1 above)
Comment (spaceisfun) [reply src/Service/ServiceExe.cs:1908]: #23

**Review by hajya — COMMENTED** [review]
Consider going back and looking the rest of your change/code for the feedback Nuri gave. otherwise looks great!

**Review by hajya — APPROVED** [review]

**Pull request merged**